packages/tools/client-debugger/client-debugger-chrome-extension/src/devtools/BackgroundConnection.ts

```diff
@@ -89,9 +89,11 @@ export class BackgroundConnection
 	/**
 	 * Post message to Background Script.
 	 */
-	public postMessage(message: IDebuggerMessage): void {
+	public postMessage(message: IBaseDebuggerMessage): void {
+		const sourcedMessage: IDebuggerMessage = message as IDebuggerMessage;
```

Comment:
Nit: this could be expressed via the following to be atomic:
```typescript
const sourcedMessage: IDebuggerMessage = {
    ...message,
    source,
};
```